### ISSUE TITLE
请教API问题。

### ISSUE THREAD

**Issue by hahadaba** [You2php/you2php#198]
请教站长，我想要得到某个类别下面、根据关键词搜索的所有内容（返回JSON最佳），该如何使用呢？
举例：类别为cont=category&sortid=15
关键词：仓鼠

非常感谢！

**Comment by zxq2233** [comment]
`https://www.googleapis.com/youtube/v3/search?part=snippet&q=bieber&type=video&videoCategoryId=10&key={YOUR_API_KEY}`

具体参考文档
https://developers.google.com/youtube/v3/docs/search

**Comment by zxq2233** [comment]
分类id这里有
https://gist.github.com/dgp/1b24bf2961521bd75d6c

**Comment by hahadaba** [comment]
是的。按照总量7500个记录、每页50条，那么应该有250页。但是我取到12页左右，返回的json里面，items就是空值了。

**Comment by hahadaba** [comment]
没有解决上述问题，可能是google不允许读取太多记录。
不过已经够用了。
我写好了一个微信小程序，关于仓鼠视频的，能够自动更新中文、日语、韩语、英语的记录，有兴趣的可以玩一下，叫做“仓鼠大师”。
方法是：打开微信 -> 发现 -> 小程序 -> 搜索 -> 仓鼠大师

**Comment by zxq2233** [comment]
应该不是限制，api本来就是拿来调用的，你去https://stackoverflow.com/
问一下

**Comment by hahadaba** [comment]
如果搜索关键字中间有空格，该如何处理？如bon jovi
目前有空格的关键字返回一律为空。
搜索也没有找到说明。

**Comment by zxq2233** [comment]
@hahadaba urlencode编码

**Comment by hahadaba** [comment]
用urlencode不行。
把空格替换成%20就解决了。

**Comment by zxq2233** [comment]
@hahadaba 其实就是加个“+”加号

**Comment by hahadaba** [comment]
在某些在页面watch时有个问题，视频不能显示，错误信息是：the media couold not be loaded,eigher because the server or network failed or because the format is not supported
但是在www.youtube.com网址观看则无问题。
如：https://www.youtube.com/watch?v=**reeNUvkM1ZA**
站长请检查一下，是不是lib某处的处理不够完善？

**Comment by fixlin** [comment]
打开网页，为什么网速那么慢，只有几kb，不定时的跳到一百多kb，视频没法正常观看？求指教
